### PR TITLE
Type documentation fixes for Doc

### PR DESCRIPTION
**Pull request by svlandeg**
## Description

- Fix `doc.pyi` to include the fact that `char_span()` can return `None`
- Unify type documentation of `exclude`: it's a `tuple` by default, originally documented to be a `List`. Can be documented as an `Iterable[str]`, like we do for `util.to_dict` and the `to_disk` method in `doc.pyi`.

### Types of change
docs fixes

## Checklist
- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/tokens/doc.pyx [hunk 1]
```diff
@@ -1361,11 +1361,9 @@ cdef class Doc:
     def to_dict(self, *, exclude=tuple()):
         """Export the document contents to a dictionary for serialization.
 
-        exclude (list): String names of serialization fields to exclude.
+        exclude (Iterable[str]): String names of serialization fields to exclude.
         RETURNS (bytes): A losslessly serialized copy of the `Doc`, including
             all annotations.
-
-        DOCS: https://spacy.io/api/doc#to_bytes
```
Comment (svlandeg) [inline spacy/tokens/doc.pyx:1368]: This link was wrong, this method is currently not documented. Alternatively, we could go ahead and document it and fix this link. Not sure whether this needs to be in the public API though.

Comment (adrianeboyd) [reply spacy/tokens/doc.pyx:1368]: I'm confused, this is part of the public API and is here: https://spacy.io/api/doc#to_bytes

File: spacy/tokens/doc.pyx [hunk 2]
```diff
@@ -1361,11 +1361,9 @@ cdef class Doc:
     def to_dict(self, *, exclude=tuple()):
         """Export the document contents to a dictionary for serialization.
 
-        exclude (list): String names of serialization fields to exclude.
+        exclude (Iterable[str]): String names of serialization fields to exclude.
         RETURNS (bytes): A losslessly serialized copy of the `Doc`, including
```
Comment (adrianeboyd) [inline spacy/tokens/doc.pyx:1365]:
```suggestion
        RETURNS (Dict): A losslessly serialized copy of the `Doc`, including
```

File: spacy/tokens/doc.pyi [hunk 3]
```diff
@@ -179,16 +179,12 @@ class Doc:
...
-    ) -> Doc: ...
-    def to_dict(self, *, exclude: Union[List[str], Tuple[str]] = ...) -> bytes: ...
-    def from_dict(
-        self, msg: bytes, *, exclude: Union[List[str], Tuple[str]] = ...
+        self, path: Union[str, Path], *, exclude: Iterable[str] = ...
     ) -> Doc: ...
+    def to_bytes(self, *, exclude: Iterable[str] = ...) -> bytes: ...
+    def from_bytes(self, bytes_data: bytes, *, exclude: Iterable[str] = ...) -> Doc: ...
+    def to_dict(self, *, exclude: Iterable[str] = ...) -> bytes: ...
+    def from_dict(self, msg: bytes, *, exclude: Iterable[str] = ...) -> Doc: ...
```
Comment (adrianeboyd) [inline spacy/tokens/doc.pyi:187]:
```suggestion
    def to_dict(self, *, exclude: Iterable[str] = ...) -> Dict: ...
    def from_dict(self, msg: Dict, *, exclude: Iterable[str] = ...) -> Doc: ...
```

File: spacy/tokens/doc.pyx (hunk 2 above)
Comment (adrianeboyd) [reply spacy/tokens/doc.pyx:1365]: `from_dict` needs the same change, but I can't propose an edit here.

**Pull request merged**